src/main/java/com/amazonaws/services/sqs/AmazonSQSVirtualQueuesClient.java

```diff
@@ -288,7 +288,13 @@ public HostQueue(String queueUrl) {
         }
 
         private void dispatchMessage(Message message) {
-            String queueName = message.getMessageAttributes().get(VIRTUAL_QUEUE_NAME_ATTRIBUTE).getStringValue();
+            MessageAttributeValue messageAttributeValue = message.getMessageAttributes().get(VIRTUAL_QUEUE_NAME_ATTRIBUTE);
+            // Case where a message was sent with missing attribute __AmazonSQSVirtualQueuesClient.QueueName
+            if (messageAttributeValue == null) {
```

Comment:
Can we add the `virtualQueue` null check here as well?

Reply:
We have a `virtualQueue` null check after this, since we need to use the `messageAttributeValue ` to get the `virtualQueue`